feat: add commitizen cli and commit standards (#340)
cytomining/pycytominer: CONTRIBUTING.md

### Git commit messages

Pycytominer uses [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard for commit messages to aid in automatic changelog generation.

> The easiest way to prepare commit messages that follow this standard is to use [commitizen](https://commitizen-tools.github.io/commitizen/), which comes pre-installed in the devcontainer.

Reviewer: Minor suggestion to move away from subjective language which might change / have multiple perspectives.
```suggestion
We prepare commit messages that follow this standard using [commitizen](https://commitizen-tools.github.io/commitizen/), which comes pre-installed in the devcontainer.
```

Author: Accepted

> A pre-push commit hook will also run automatically to ensure that your commit messages follow this standard.

Reviewer: ```suggestion
A pre-push commit hook from [pre-commit](https://pre-commit.com/) may be used to automatically ensure your commit messages follow this standard.
```

Author: Removing the text about the pre-commit hook so we can discuss it further after 1.0 release.

### Python style guide

For python code style, we use [black](https://github.com/psf/black).
Please use black before committing any code.
We will not accept code contributions that do not use black.

> If you have set up your development environment using one of the devcontainer options specified above, the containers will install all required formatting tools, which will run automatically on any modified files before commits (using a tool called [pre-commit](https://pre-commit.com/)).

Reviewer: Suggestion for simplifications.
```suggestion
The devcontainer will install all required formatting tools.
Formatting checks in devcontainers run automatically on any modified files before commits (using a tool called [pre-commit](https://pre-commit.com/)).
```

Author: Accepted.

Author: Revised this language to be dev environment agnostic.

> 

Reviewer: What happens if someone isn't using the devcontainer? Should we make a special mention here to make sure they have an equitable chance to automatically format things with a small mention here?

Author: Good point. I rewrote the suggestion to be agnostic to dev environment.
